### PR TITLE
Kristian's Homework 2 Memory V2

### PR DESCRIPTION
**Pull request by KristianLindberg**
Kristian's Homework 2 Memory V2

### CODE REVIEW

**Comment by KristianLindberg** [comment]
Kristian's completed Memory ver.3 exersice.

File: script.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+var nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15].sort(function() {
+  return .5 - Math.random();
+});
```
Comment (afeld) [inline script.js:3]: We talked about this a bit in-person, but how could you avoid a) hard-coding all of these numbers, and b) making the number of values correspond to the `x` and `y` specified below?

File: script.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+function table (x, y){
```
Comment (afeld) [inline script.js:5]: What do these arguments correspond to? (Please avoid using single-letter variable names!)

File: script.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      document.write("<td class=\"" + newClass + "\">" + (" ? ") + "</td>");
+    }
+    document.write("</tr>");  
+  }
+  document.write("</table>");
+};
+
+table (5, 6); 
+
+var boardIntel = function(theId){
```
Comment (afeld) [inline script.js:24]: Minor: The convention for classes is to use TitleCase.

File: script.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  this.clicked = theId;
+  boardIntel.group = [];
```
Comment (afeld) [inline script.js:26]: Why use a property on the class rather than the instance?

File: script.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+};
+
+boardIntel.prototype.pusher = function(){
```
Comment (afeld) [inline script.js:29]: I don't think this method name is really representative of what it's doing...what would be a more descriptive name?

File: script.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+     tdId = this.className;
+    this.innerHTML = tdId;
+    boardIntel.group.push(tdId);
+    console.log(boardIntel.group)
+    boardIntel2.matcher();
+  });
+};
+
+boardIntel.prototype.matcher = function(){
+  var x = boardIntel.group;
```
Comment (afeld) [inline script.js:40]: Ditto about the single-letter variable names!

File: script.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  if (x[x.length-1] ==  x[x.length-2]){
```
Comment (afeld) [inline script.js:41]: What is this checking? Moving the conditional to a well-named method will make it more obvious.

File: script.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    var firstTd = x[x.length-1];
+    var secondTd= x[x.length-2];
```
Comment (afeld) [inline script.js:43]: Might want to move these lines up, so that you can use the variables for the conditional.

**Comment by afeld** [comment]
Good start, though the gameplay still needs a bit of work:
- After I reveal two tiles, they should flip back over (turn into `?`s).
- Clicking any card twice gives me a "You win!" and removes those tiles, even if the other one wasn't revealed.

Also, please fix [the JSHint warnings](https://travis-ci.org/advanced-js/memory/builds/89533600). Thanks!

**Pull request closed without merging**